### ISSUE TITLE
`@arcgis/core@4.20` throws fatal errors during OAuth 2.0 routine in `@angular/core@12.2+` production build

### ISSUE THREAD

**Issue by vitale232** [Esri/jsapi-resources#329]
I am reporting an issue with 

- [ ] TypeScript definitions
- [x] another resource in this repository

# Description

I'm using the Esri ArcGIS JavaScript API to implement OAuth 2.0 against AGOL in my Angular 12.2 application. My implementation works when using `ng serve`, but it throws fatal errors (the user never gets logged in) when running the `ng build` output. The issue is present both locally and when deployed to Azure.

When using `@angular/core@12.1`, the `ng build` output works as expected.

## Replication

[I've forked this repository](https://github.com/vitale232/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli) and adjusted the Angular sample to better mimic my application structure. _Note: You likely need to adjust the `portalUrl` property that is set in the `environment.ts` files to point to your organization's AGOL._

On the `master` branch, run `ng build`. Then, run the production build locally on port 3000 (I use `lite-server` from `npm`). When you attempt to login, you'll see the following error in the console:

```
ERROR Error: Uncaught (in promise): TypeError: Cannot convert undefined or null to object
TypeError: Cannot convert undefined or null to object
    at propertyIsEnumerable (<anonymous>)
    at Object.15399 (556.5a938027116850e3dd9a.js:1)
    at f (runtime.33b405900dbd97a9f460.js:1)
    at Object.97019 (556.5a938027116850e3dd9a.js:1)
    at f (runtime.33b405900dbd97a9f460.js:1)
    at Object.556 (556.5a938027116850e3dd9a.js:1)
    at f (runtime.33b405900dbd97a9f460.js:1)
    at Object.36526 (5455.2ac6ede174dbbcfafe2f.js:1)
    at f (runtime.33b405900dbd97a9f460.js:1)
    at Module.89217 (6638.55e8b1ac1c5c3b03a94e.js:1)
    at G (polyfills.c17f96e21416c50b0eda.js:1)
    at G (polyfills.c17f96e21416c50b0eda.js:1)
    at polyfills.c17f96e21416c50b0eda.js:1
    at T.invokeTask (polyfills.c17f96e21416c50b0eda.js:1)
    at Object.onInvokeTask (main.8c73020d17b9bcc32279.js:1)
    at T.invokeTask (polyfills.c17f96e21416c50b0eda.js:1)
    at I.runTask (polyfills.c17f96e21416c50b0eda.js:1)
    at L (polyfills.c17f96e21416c50b0eda.js:1)
```

Interesting, the replication throws a different error than my application where I originally discovered this issue. In my actual application, this is the error I receive:

```
polyfills.88b245640e63b0722765.js:1 TypeError: Property description must be an object: undefined
    at defineProperty (<anonymous>)
    at O._fetchSelf (7502.4509da83417893d1ee2b.js:1)
    at 7502.4509da83417893d1ee2b.js:1
    at T.invoke (polyfills.88b245640e63b0722765.js:1)
    at Object.onInvoke (main.a45f9ee8e967c2f65b80.js:1)
    at T.invoke (polyfills.88b245640e63b0722765.js:1)
    at I.run (polyfills.88b245640e63b0722765.js:1)
    at polyfills.88b245640e63b0722765.js:1
    at T.invokeTask (polyfills.88b245640e63b0722765.js:1)
    at Object.onInvokeTask (main.a45f9ee8e967c2f65b80.js:1)
```

## Workaround

By installing Angular 12.1 in lieu the most recent version, I can workaround this issue. In the linked repository, there is a branch called `oauth-ng12.1`. If you switch to that branch, run `npm install`, then `ng build`, the output of the build successfully authenticates.

**Comment by andygup** [comment]
@vitale232 that is correct, this is a known issue with Angular 12.2.x and we've documented it here: https://github.com/Esri/jsapi-resources/tree/master/esm-samples/jsapi-angular-cli#known-issues. The good news is the bug seemed to be fixed at 13.0.0-next. 

Cross-reference: https://github.com/Esri/feedback-js-api-next/issues/131#issuecomment-897211321

**Comment by vitale232** [comment]
Thanks @andygup... I guess that's why you should always read the README ;-)
